server/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -445,6 +445,9 @@ public Scroll scroll() {
      * If set, will enable scrolling of the search request.
      */
     public SearchRequest scroll(Scroll scroll) {
+        if (scroll != null && this.source != null && this.source.from() > 0) {
+            throw new IllegalArgumentException("scroll cannot be used with a search using [from]");
+        }
```

Comment:
This solution still leaves room for e.g. 
```
SearchRequest searchRequest = createSearchRequest().source(new SearchSourceBuilder().size(0));
searchRequest.scroll(new Scroll(TimeValue.timeValueSeconds(10)));
searchRequest.source().from(1);
```
to create an "invalid" request that will later throw exception on validation.